pants-plugins/uses_services/rabbitmq_rules.py

```diff
@@ -0,0 +1,178 @@
+        if not target.has_field(UsesServicesField):
+            return False
+        uses = target.get(UsesServicesField).value
+        return uses is not None and "rabbitmq" in uses
+
+
+@rule(
+    desc="Ensure rabbitmq is running and accessible before running tests.",
+    level=LogLevel.DEBUG,
+)
+async def rabbitmq_is_running_for_pytest(
+    request: PytestUsesRabbitMQRequest,
+) -> PytestPluginSetup:
+    # this will raise an error if rabbitmq is not running
+    _ = await Get(RabbitMQIsRunning, UsesRabbitMQRequest())
+
+    return PytestPluginSetup()
+
+
+@rule(
+    desc="Test to see if rabbitmq is running and accessible.",
+    level=LogLevel.DEBUG,
+)
+async def rabbitmq_is_running(
+    request: UsesRabbitMQRequest, platform: Platform
+) -> RabbitMQIsRunning:
+    script_path = "./is_rabbitmq_running.py"
+
+    # pants is already watching this directory as it is under a source root.
+    # So, we don't need to double watch with PathGlobs, just open it.
+    with open(is_rabbitmq_running_full_path, "rb") as script_file:
+        script_contents = script_file.read()
+
+    script_digest, kombu_pex = await MultiGet(
+        Get(Digest, CreateDigest([FileContent(script_path, script_contents)])),
+        Get(
+            VenvPex,
+            PexRequest(
+                output_filename="kombu.pex",
+                internal_only=True,
+                requirements=PexRequirements({"kombu",}),
+            ),
+        ),
+    )
+
+    result = await Get(
+        FallibleProcessResult,
+        VenvPexProcess(
+            kombu_pex,
+            argv=(
+                script_path,
+                "amqp://guest:guest@127.0.0.1:5672/",
+            ),
+            input_digest=script_digest,
+            description="Checking to see if RabbitMQ is up and accessible.",
+            # this can change from run to run, so don't cache results.
+            cache_scope=ProcessCacheScope.PER_SESSION,
+            level=LogLevel.DEBUG,
+        ),
+    )
+    is_running = result.exit_code == 0
+
+    if is_running:
+        return RabbitMQIsRunning()
+
+    # rabbitmq is not running, so raise an error with instructions.
+    raise ServiceMissingError(
+        platform,
+        ServiceSpecificMessages(
+            service="rabbitmq",
+            service_start_cmd_el_7="service rabbitmq-server start",
+            service_start_cmd_el="systemctl start rabbitmq-server",
+            not_installed_clause_el="this is one way to install it:",
+            install_instructions_el=dedent(
+                """\
+                # Add key and repo for erlang and RabbitMQ
+                curl -sL https://packagecloud.io/install/repositories/rabbitmq/erlang/script.rpm.sh | sudo bash
+                curl -sL https://packagecloud.io/install/repositories/rabbitmq/rabbitmq-server/script.rpm.sh | sudo bash
+                sudo yum makecache -y --disablerepo='*' --enablerepo='rabbitmq_rabbitmq-server'
```

Comment:
Given the fun we have with RabbitMQ and the fact that we often have to choose specific versions, I wonder if in these commands we should put a comment to say to refer to the appropriate StackStorm manual installation instructions for your release on ho to install RabbitMQ locally.
That way if we change we don't need to update the instructions, we can just let them to read our manual instructions.

Reply:
I looked at that, but the scripts in our docs combine installing all of the dependent services: Mongo, rabbitmq, and redis ... So, it's hard to call out just what's required for any one service.

For deb (below), the instructions were so complex (both upstream and in our manual install), I opted to just link to the upstream docs. Plus, the upstream had a copy/pastable quickstart script in the docs that makes the install fairly simple.

Sadly, they don't provide an equivalent quickstart script for rpm. But we could still link to the section of the docs: https://www.rabbitmq.com/install-rpm.html#cloudsmith

So, this is basically the instructions extracted from our docs, filtered down to only deal with rabbitmq. Is there a cleaner place to link to instead of embedding this here?

Reply:
YEah you are right we don't have the dependencies separated. Maybe a comment stating to check our documentation in case we put any constraints on rabbitmq or erlang versioning.

e.g. if we move back to needing to constrain the erlang version again, then we're going to remember to change the st2docs and the automated installs, but not this file.